tests/core/full_node/ram_db.py

```diff
@@ -1,21 +1,24 @@
 from typing import Tuple
 from pathlib import Path
 
+import random
 import aiosqlite
 
 from chia.consensus.blockchain import Blockchain
 from chia.consensus.constants import ConsensusConstants
 from chia.full_node.block_store import BlockStore
 from chia.full_node.coin_store import CoinStore
 from chia.full_node.hint_store import HintStore
-from chia.util.db_wrapper import DBWrapper
+from chia.util.db_wrapper import DBWrapper2
 
 
-async def create_ram_blockchain(consensus_constants: ConsensusConstants) -> Tuple[aiosqlite.Connection, Blockchain]:
-    connection = await aiosqlite.connect(":memory:")
-    db_wrapper = DBWrapper(connection)
+async def create_ram_blockchain(consensus_constants: ConsensusConstants) -> Tuple[DBWrapper2, Blockchain]:
+    uri = f"file:db_{random.randint(0, 99999999)}?mode=memory&cache=shared"
```

Comment:
this isn't actually creating a file. the `?mode=memory` means it's a memory-only database. The name created here is just to allow multiple connections to the same memory-only DB (otherwise each connection will create its own in-memory database)

Reply:
Quite so, thanks.